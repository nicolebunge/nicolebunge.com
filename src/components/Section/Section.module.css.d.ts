declare const styles: {
  readonly section: string;
  readonly 'section--full': string;
};
export = styles;
