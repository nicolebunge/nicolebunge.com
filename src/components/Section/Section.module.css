.section {
  align-items: center;
  display: flex;
  padding-bottom: var(--space-xxl);
  padding-top: var(--space-xxl);
}

@media (width >= 1080px) {
  .section {
    padding-bottom: var(--space-xxxl);
    padding-top: var(--space-xxxl);
  }
}

.section--full {
  min-height: 100vh;
}
