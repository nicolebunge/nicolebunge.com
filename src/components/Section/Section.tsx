import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import { section, sectionFull } from './Section.module.css';

interface SectionProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
  isFull?: boolean;
}

function Section(props: SectionProps): JSX.Element {
  const { className, isFull = false, ...otherProps } = props;

  return (
    <section
      className={clsx(className, section, {
        [sectionFull]: isFull,
      })}
      {...otherProps}
    />
  );
}

export type { SectionProps };
export default Section;
