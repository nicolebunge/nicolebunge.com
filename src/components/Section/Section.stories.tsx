import Section from '.';

const meta = {
  component: Section,
};

const Default = {
  args: {
    children: 'Section',
  },
};

export { Default };
export default meta;
