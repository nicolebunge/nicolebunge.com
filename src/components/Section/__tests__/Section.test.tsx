import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Section.stories';

const { Default } = composeStories(stories);

describe('Section component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
