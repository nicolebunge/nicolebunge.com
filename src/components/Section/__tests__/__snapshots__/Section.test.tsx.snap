// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section component > renders correctly 1`] = `
<section
  class="section"
>
  Section
</section>
`;
